### PR TITLE
Add `vllm serve` to wrap `vllm.entrypoints.openai.api_server`

### PR DESCRIPTION
**Pull request by simon-mo**
Easier to type. It will be now 

```
(base) xmo@simon-devbox:~/vllm$ vllm serve --help
usage: vllm serve <model_tag> [options]

positional arguments:
  model                 The model tag to serve

options:
  -h, --help            show this help message and exit
  --host HOST           host name
  --port PORT           port number
  --uvicorn-log-level {debug,info,warning,error,critical,trace}
                        log level for uvicorn
  --allow-credentials   allow credentials
  --allowed-origins ALLOWED_ORIGINS
                        allowed origins
  --allowed-methods ALLOWED_METHODS
                        allowed methods
.....
```

### CODE REVIEW

File: benchmarks/benchmark_serving.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 On the server side, run one of the following commands:
     vLLM OpenAI API server
-    python -m vllm.entrypoints.openai.api_server \
-        --model <your_model> --swap-space 16 \
+    vllm serve <your_model> \
```
Comment (rkooo567) [inline benchmarks/benchmark_serving.py:5]: Q for backward compatibility: previous command still works?

File: vllm/entrypoints/openai/api_server.py [hunk 2]
```diff
@@ -26,6 +26,8 @@
 
 TIMEOUT_KEEP_ALIVE = 5  # seconds
 
+engine: AsyncLLMEngine = None
```
Comment (rkooo567) [inline vllm/entrypoints/openai/api_server.py:29]: shoudln't this be optional?

File: vllm/entrypoints/openai/api_server.py [hunk 3]
```diff
@@ -173,3 +178,11 @@ async def authentication(request: Request, call_next):
                 ssl_certfile=args.ssl_certfile,
                 ssl_ca_certs=args.ssl_ca_certs,
                 ssl_cert_reqs=args.ssl_cert_reqs)
+
+
+if __name__ == "__main__":
+    # NOTE(simon):
+    # This section should be in sync with vllm/scripts.py for CLI entrypoints.
```
Comment (rkooo567) [inline vllm/entrypoints/openai/api_server.py:185]: any way to add a simple regression test for this?

File: vllm/scripts.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    parser = argparse.ArgumentParser(description="vLLM CLI")
+    subparsers = parser.add_subparsers()
+
+    serve_parser = subparsers.add_parser(
+        "serve",
+        help="Start the vLLM OpenAI Compatible API server",
+        usage="vllm serve <model_tag> [options]")
+    make_arg_parser(serve_parser)
+    # Override the `--model` optional argument, make it positional.
+    serve_parser.add_argument("model", type=str, help="The model tag to serve")
```
Comment (rkooo567) [inline vllm/scripts.py:18]: what's happening if vllm serve <model> --model <model2>?

File: vllm/scripts.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    serve_parser.set_defaults(func=run_server)
+
+    args = parser.parse_args()
+    if hasattr(args, "func"):
```
Comment (rkooo567) [inline vllm/scripts.py:22]: this part of code is confusing. Add a comment to explain what this does?

**Review by rkooo567 — APPROVED** [review]

File: benchmarks/benchmark_serving.py (hunk 1 above)
Comment (simon-mo) [reply benchmarks/benchmark_serving.py:5]: yup works exactly as is.

**Comment by DarkLight1337** [comment]
As mentioned in #4709, the `:prog:` value under CLI args should be updated to `vllm serve`.

File: docs/source/serving/openai_compatible_server.md [hunk 6]
```diff
@@ -95,8 +95,7 @@ template, or the template in string form. Without a chat template, the server wi
 and all chat requests will error.
 
 ```bash
-python -m vllm.entrypoints.openai.api_server \
-  --model ... \
+vllm serve ... \
   --chat-template ./path-to-chat-template.jinja
 ```
```
Comment (DarkLight1337) [inline docs/source/serving/openai_compatible_server.md:101]: Based on #4709, the `:prog:` value under CLI args (line 111) should be updated to `vllm serve`.

**Review by DarkLight1337 — CHANGES_REQUESTED** [review]

**Comment by sasha0552** [comment]
Is there any update on this? Having the command simplifies the installation of `vllm` to
```sh
pipx install vllm

# now you can use the command
vllm serve --help
```
with the wonderful [pipx](https://github.com/pypa/pipx) tool that manages virtual environments automatically. In the current state you cannot use `vllm` with `pipx`, `pipx` only supports exposing commands.

**Comment by DarkLight1337** [comment]
Would be nice if #4794 is also made available via CLI.

File: vllm/scripts.py (hunk 4 above)
Comment (EthanqX) [reply vllm/scripts.py:18]: renamed the positional arg to be `model-tag` to avoid namespace conflict

**Comment by EthanqX** [comment]
Please refer to #5090 for the complete new CLI.

**Pull request closed without merging**